students/brian_minsk/lesson05/assignment/database.py

```diff
@@ -0,0 +1,177 @@
+""" Code for lesson05 assignment.
+"""
+
+import csv
+import logging
+from pymongo import MongoClient
+
+logging.basicConfig(level=logging.ERROR)
+LOGGER = logging.getLogger(__name__)
+
+class MongoDBConnection():
+    """ MongoDB Connection
+
+    This class's code is swiped from "Part 5: Python code" from the lesson05 materials.
+    """
+
+    def __init__(self, host='127.0.0.1', port=27017):
+        """ be sure to use the ip address not name for local windows"""
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+MONGO = MongoDBConnection()
+with MONGO:
+    DB = MONGO.connection.media
+    PRODUCT_COLLECTION = DB['product']
+    CUSTOMER_COLLECTION = DB['customers']
+    RENTAL_COLLECTION = DB['rentals']
+
+def import_data(directory_name="csv_files", product_file="product.csv",
+                customer_file="customers.csv", rental_file="rentals.csv"):
+    """ Take a directory name and three csv files as input, one with product data, one with
+    customer data, and the third one with rentals data. Create and populate a new MongoDB
+    database with these data. Returns 2 tuples: the first with a document count of the number of
+    products, customers and rentals added (in that order), the second with a count of any
+    errors that occurred, in the same order.
+
+    Keyword arguments:
+    directory_name - the directory that contains the csv_files
+    product_file - a csv file with product data with the first line containing the field names
+    customer_file - a csv file with customer data with the first line containing the field
+                    names
+    rentals_file - a csv file with rental data with the first line containing the field names
+    """
+    n_products, errs_products = populate_collection(directory_name + '\\' + product_file,
```

Comment:
The '\\' in these lines of code causes your unit tests to fail on my Mac, because of the different conventions on Unix-like systems, but it was easy for me to fix it for reviewing your assignment.

You might need to consider this in some of your future projects need to work on both Windows and Unix-like operating systems. That said, no need to make any changes for this assignment :-)

Reply:
Gotcha. I did that quickly and meant to go back and use the Python file path creation functions but I forgot... Thanks for fixing.